### PR TITLE
[VAN-1395][VAN-671] enable simplification pass on extracted body functions

### PR DESCRIPTION
**Pull request by tim-hoffman**

### CODE REVIEW

File: circom/tests/calls/arena_missing_val.circom [hunk 1]
```diff
@@ -49,21 +49,11 @@ component main = BigMod();
...
 //CHECK-NEXT: if.then:
 //CHECK-NEXT:   %[[T004:[0-9a-zA-Z_\.]+]] = getelementptr i256, i256* %var_2, i32 0
-//CHECK-NEXT:   %[[T005:[0-9a-zA-Z_\.]+]] = getelementptr i256, i256* %var_3, i32 0
-//CHECK-NEXT:   %[[T006:[0-9a-zA-Z_\.]+]] = load i256, i256* %[[T005]], align 4
-//CHECK-NEXT:   %[[T007:[0-9a-zA-Z_\.]+]] = getelementptr i256, i256* %var_4, i32 0
-//CHECK-NEXT:   %[[T008:[0-9a-zA-Z_\.]+]] = load i256, i256* %[[T007]], align 4
-//CHECK-NEXT:   %[[T992:[0-9a-zA-Z_\.]+]] = call i256 @fr_sub(i256 %[[T006]], i256 %[[T008]])
-//CHECK-NEXT:   store i256 %[[T992]], i256* %[[T004]], align 4
+//CHECK-NEXT:   store i256 0, i256* %[[T004]], align 4
 //CHECK-NEXT:   br label %if.merge
```
Comment (0xddom) [inline circom/tests/calls/arena_missing_val.circom:57]: I don't have empirical data to back up my claim but those branches (even if tautological) may affect SVF's analysis. Can we remove them if we know for sure they are tautos? (Doesn't have to be in this PR)

**Comment by tim-hoffman** [comment]
Only the final commit here is new. I guess I merged too soon so it listed them all.

**Review by 0xddom — APPROVED** [review]
neato

File: circom/tests/calls/arena_missing_val.circom (hunk 1 above)
Comment (tim-hoffman) [reply circom/tests/calls/arena_missing_val.circom:57]: Yeah, I have a ticket to look into this. https://veridise.atlassian.net/browse/VAN-1393

**Pull request merged**